docs/USAGE-GUIDE.md

```diff
@@ -791,9 +796,19 @@ Authentication is usually the most expensive part of the handshake. To avoid the
 
 ### Configuring the Trust Store
 
-To validate the peer’s certificate, the local “trust store” must contain a certificate that can authenticate the peer’s certificate.
+To validate the peer’s certificate, a config's local “trust store” must contain a certificate
+that can authenticate the peer’s certificate. To add certificates to the trust store, call
+`s2n_config_set_verification_ca_location()` or `s2n_config_add_pem_to_trust_store()`.
+
+Operating systems typically install a set of default system certificates used by TLS clients to
+verify the authenticity of public TLS servers. `s2n_config_new()` initializes the trust store with
+these system certificates by default. If s2n-tls is verifying certificates and is not a TLS client
+connecting to a public TLS server, the system certificates should be cleared with
+`s2n_config_wipe_trust_store()` before calling `s2n_config_set_verification_ca_location()` or
+`s2n_config_add_pem_to_trust_store()`.
```

Comment:
This is a little tricky. The system trust store might not just contain like generic public CA certificates. Since the system trust store can be configured, it might contain custom or organization-specific certificates. So I'm not sure about the first sentence saying "Operating systems typically install", and customer might need to keep the trust store even if not accessing a public internet server.
```suggestion
`s2n_config_new()` initializes the trust store with the default system certificates, which may include public CAs or other unexpected certificates. If s2n-tls is verifying certificates and but does not require the default system certificates, they should be cleared with
`s2n_config_wipe_trust_store()` before calling `s2n_config_set_verification_ca_location()` or
`s2n_config_add_pem_to_trust_store()`.
```

Reply:
This might also be the right place to mention the importance of verify_host_fn ;)